link_traffic/munin/link_traff2json_

```diff
@@ -0,0 +1,159 @@
+        time_mon_st = current_time 
+    else:
+        sys.stderr.write("Reading from file\n")
+        with open(data_temp_file, "r") as f:
+            data = f.read()
+        data = data.split("\n")
+        time_mon_st = last_time 
+
+    return {'data' : data, 'time_st': time_mon_st}
+#=======================================================================
+"""!@fn loadLinkConfig
+  @brief function that loads links configuration, color, name,
+   link number etc.
+  @return list of dictionaries with link's parameters """
+def loadLinkConfig():
+   with open(CONFIG_FILE, 'rb') as f:
+      reader = csv.reader(f)
+      try:
+         config = []
+         for rows in reader:
+            config.append({'num': rows[0],
+                            'name': rows[1],
+                            'ur_field': rows[2], 
+                            'color': rows[3]
+                           })
+         return config
+      except csv.Error as e:
+         sys.exit('file %s, line %d: %s' % (CONFIG_FILE, reader.line_num, e))
+#=======================================================================
+"""! @fn getLineNum @brief function tries to find link in config file 
+      based on its name 
+     @return link's number"""
+def getLineNum( lineName ):
+   configData = loadLinkConfig()
+   for items in configData:
+      if items['name'] == lineName:
+         return items['num']
+#=======================================================================
+def getTimeInfo(mon_time):
+   dictCurr = {} #!dictionary for current time
+   dictMon = {}  #!dictionary for monitoring time
+   time_st = time.time()
+   curr_local = time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(time_st)) 
+   mon_local = time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(mon_time)) 
+   dictCurr = {'localtime' : curr_local, 'timestamp'  : time_st}
+   dictMon = {'interval_sec' : interval,
+              'localtime' : mon_local,
+              'timestamp' : mon_time
+             }
+   return {'current' : dictCurr,
+           'monitoring' : dictMon
+          }
+#=======================================================================
+"""!@fn outputToJson
+  @brief function takes configuration list of lines and data
+   from socket and converts it
+   to json format and prints to output 
+   @return dict in json format """
+def outputToJson():
+   config = loadLinkConfig()
+   dataRec = getData()
+   data = dataRec['data']
+   mon_time = dataRec['time_st']
+   head = data[0].split(',')
+   vals = data[1].split(',')
+   if len(head) != len(vals):
+      sys.stderr.write("wrong number of columns!!!\n")
+      sys.exit(1)
+   data = zip(head, vals)
+   data.sort()
+   dataDict = {}
+   timeDict = getTimeInfo(mon_time)
+
+   # dict init name of link and its number 
+   for key, val in data:
+      ln = getLineNum(key.split('-')[0])
+      dataDict[ln] = {"link" : ln, "link name" : key.split('-')[0]}
+   # add the rest of values
+   for key, val in data:
```

Comment:
I don't understand why is this split into two loops (key.split and GetLineNum is called twice)